src/Fixer/Basic/PsrAutoloadingFixer.php

```diff
@@ -103,7 +103,7 @@ public function supports(\SplFileInfo $file): bool
             // ignore file with extension other than php
             ('php' !== $file->getExtension())
             // ignore file with name that cannot be a class name
-            || 0 === Preg::match('/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/', $file->getBasename('.php'))
+            || 0 === Preg::match('/^(?!\d)[\dA-Z_a-z\x80-\xff]+$/', $file->getBasename('.php'))
```

Comment:
It makes no sense to me - why would you use negative lookahead for `\d` and include it later in the matching group? Looking at the comment above numbers are totally valid chars for class names.

Reply:
This should be correct, class name (identifier in general) cannot start with a number.